src/wazuh_db/wdb_parser.c

```diff
@@ -3848,3 +3866,87 @@ int wdb_parse_mitre_get(wdb_t * wdb, char * input, char * output) {
         return -1;
     }
 }
+
+int wdb_parse_global_get_agent_labels(wdb_t * wdb, char * input, char * output) {
+    int agent_id = 0;
+    cJSON *labels = NULL;
+    char *out = NULL;
+
+    agent_id = atoi(input);
+
+    if (labels = wdb_global_get_agent_labels(wdb, agent_id), !labels) {
+        mdebug1("Error getting agent labels from global.db.");
+        snprintf(output, OS_MAXSTR + 1, "err Error getting agent labels from global.db.");
+        return OS_INVALID;
+    }
+
+    out = cJSON_PrintUnformatted(labels);
+    snprintf(output, OS_MAXSTR + 1, "ok %s", out);
+    os_free(out);
+    cJSON_Delete(labels);
+
+    return OS_SUCCESS;
+}
+
+int wdb_parse_global_set_agent_labels(wdb_t * wdb, char * input, char * output) {
+    char *next = NULL;
+    char *labels = NULL;
+    char *label = NULL;
+    char *value = NULL;
+    cJSON *data = NULL;
+    char *out = NULL;
+    char sdelim[] = { '\n', '\0' };
+
+    if (next = wstr_chr(input, ' '), !next) {
+        mdebug1("Invalid DB query syntax.");
+        mdebug2("DB query error near: %s", input);
+        snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", input);
+        return OS_INVALID;
+    }
+    *next++ = '\0';
+
+    int agent_id = atoi(input);
+
+    if (!next) {
+        mdebug1("Global DB Invalid query syntax.");
+        mdebug2("Global DB query error near: %s", input);
+        snprintf(output, OS_MAXSTR + 1, "err Invalid DB query syntax, near '%.32s'", input);
+        return OS_INVALID;
+    } else {
+        // Removing old labels from the labels table
+        if (OS_SUCCESS != wdb_global_del_agent_labels(wdb, agent_id)) {
```

Comment:
We can create a new SQL query MERGE to resolve the delete and set in one unique statement.

Reply:
I was evaluating how to use a `MERGE` statement and I would like to discuss with the team whether or not is the best alternative. The thing is that we are trying to make use of `bind` to add the values to the tuple to be inserted. This allows us to work with a single tuple a time instead of a set of tuples to perform the `MERGE`.
The alternative for this is to have a string and concatenate values, but we already agreed that shouldn't be the best approach because concatenating strings and making use of `snprintf` for the SQL queries could make the daemons run into code injection.
Let's discuss this in the next daily meeting and evaluate the alternatives.